src/components/invoices/InvoiceLink.js

```diff
@@ -1,90 +1,110 @@
+import Button from '@material-ui/core/Button'
+import { withStyles } from '@material-ui/core/styles'
 
+import DatePeriods from '../../support/DatePeriods'
 
 const styles = theme => ({
   buttonLike: {
     ...theme.typography.button,
     backgroundColor: theme.palette.common.white,
     padding: theme.spacing.unit,
-    fontSize: "0.8125rem"
+    fontSize: '0.8125rem'
   }
-});
+})
 
 class InvoiceLink extends Component {
   render() {
-    const orgUnit = this.props.orgUnit;
+    const orgUnit = this.props.orgUnit
 
-    const codes = this.props.invoices.getInvoiceTypeCodes(orgUnit);
+    const codes = this.props.invoices.getInvoiceTypeCodes(orgUnit)
 
     if (codes === undefined || codes.length === 0) {
-      return null;
+      return null
     }
 
-    const invoiceTypes = this.props.invoices.getInvoiceTypes(codes);
+    const invoiceTypes = this.props.invoices.getInvoiceTypes(codes)
 
     return invoiceTypes.map(invoiceType => (
       <React.Fragment key={this.linkTo(invoiceType)}>
-        {invoiceType.frequency === "monthly" && (
+        {invoiceType.frequency === 'monthly' && (
           <React.Fragment>
             <span className={this.props.classes.buttonLike}>
               {invoiceType.name}
             </span>
 
-            {DatePeriods.split(this.props.period, "monthly").map(
+            {DatePeriods.split(this.props.period, 'monthly').map(
               monthPeriod => (
                 <Button
-                  key={invoiceType.code+"-"+monthPeriod+"-"+orgUnit.id}
-                  variant="flat"
-                  color="primary"
-                  size="small"
+                  key={invoiceType.code + '-' + monthPeriod + '-' + orgUnit.id}
+                  variant='text'
+                  color='primary'
+                  size='small'
                   component={Link}
                   to={this.linkTo(invoiceType, monthPeriod)}
                   title={monthPeriod}
                 >
-                  {DatePeriods.displayName(monthPeriod, "monthYear")}
+                  {DatePeriods.displayName(monthPeriod, 'monthYear')}
                 </Button>
               )
             )}
           </React.Fragment>
         )}
-        {invoiceType.frequency === "quarterly" && (
+        {invoiceType.frequency === 'quarterly' && (
           <Button
-            variant="flat"
-            color="primary"
-            size="small"
+            variant='text'
+            color='primary'
+            size='small'
             component={Link}
             to={this.linkTo(invoiceType, this.props.period)}
             title={this.props.period}
           >
             {invoiceType.name}
           </Button>
         )}
+        {invoiceType.frequency === 'sixMonthly' && (
+          <React.Fragment>
+            <span className={this.props.classes.buttonLike}>
+              {invoiceType.name}
+            </span>
+            {DatePeriods.splitYearQuarter(this.props.period, 'sixMonthly').map(
```

Comment:
I'd like too keep the "used" api of DatePeriods to
 - DatePeriods.split(period,  frequency)
 - DatePeriods.displayName(period, format)
all other functions should stay "private/unused"
and adapt the spec of DatePeriods if necessary.